.evergreen/config_generator/etc/distros.py

```diff
@@ -0,0 +1,296 @@
+from typing import Optional
+
+from pydantic import BaseModel, validator
+from packaging.version import Version
+
+
+class Distro(BaseModel):
+    """
+    Defines common properties of a given Evergreen distro.
+
+    * name: Name of the distro.
+    * os: Name of the operating system.
+    * os_type: One of Linux, MacOS, or Windows.
+    * os_ver: Version of the operating system.
+    * vs_ver: Version of Visual Studio available.
+    * size: Size of tasks the distro is designed to handle.
+    * arch: Target architecture.
+    """
+
+    name: str
+    os: Optional[str]
+    os_type: Optional[str]
+    os_ver: Optional[str]
+    vs_ver: Optional[str]
```

Comment:
Instead of using a validator, you can use a `typing.Literal` annotation:

```py
os_type: None | Literal['linux', 'macos', 'windows']
vs_version: None | Literal['2013', '2015', '2017', '2019', '2022', 'vsCurrent', 'vsCurrent2', 'vsMulti']
size: None | Literal['small', 'large']
# etc.
```

(May require `typing_extensions.Literal` for older Python versions)